src/type-layout.md

```diff
@@ -162,7 +163,26 @@ representation will not change the layout of `Inner`.
 Nominal types without a `repr` attribute have the default representation.
 Informally, this representation is also called the `rust` representation.
 
-There are no guarantees of data layout made by this representation.
+The only data layout guarantees made by this representation are those required
+for soundness. They are:
+
+ 1. The fields of the struct are properly aligned.
```

Comment:
This section is discussing the layout of structs, enums, and unions. However, these additions seem to be very struct-centric.  Can the wording be tweaked to encompass those different types?